core/types/types.cc

```diff
@@ -334,6 +334,21 @@ LiteralType::LiteralType(SymbolRef klass, NameRef val)
     ENFORCE(klass == Symbols::String() || klass == Symbols::Symbol());
 }
 
+int64_t LiteralType::asInteger() const {
+    ENFORCE_NO_TIMER(literalKind == LiteralTypeKind::Integer);
+    return value;
+}
+
+double LiteralType::asFloat() const {
+    ENFORCE_NO_TIMER(literalKind == LiteralTypeKind::Float);
+    return floatval;
+}
+
+core::NameRef LiteralType::asName() const {
+    ENFORCE_NO_TIMER(literalKind == LiteralTypeKind::Symbol || literalKind == LiteralTypeKind::String);
+    return NameRef(NameRef::WellKnown{}, value);
```

Comment:
This seems wrong, doesn't it? shouldn't we pass a GlobalState from the call site?

Reply:
Sure, I can do that.

Reply:
(every call site this replaced seems to have a GlobalState in scope directly)

Reply:
The only place that would be complicated by that is the pickling code, as it used to just always encode the `u8` value of the union.

Reply:
Fixed. I did have to add an `unsafe` variant for the serializer callsite.

Reply:
Gotcha, so at that point, we could pass WellKnown, and write a comment about why it doesn't matter that we're using WellKnown for that call site.